### PR TITLE
Implement popout location in show()

### PR DESCRIPTION
**Pull request by duytnguyendtn**
### Description

@mariobuikhuizen implemented the necessary methods to implement our missing show location options; this PR updates our required ipypopout version to 0.0.10 and implements our missing popout locations. Thanks @mariobuikhuizen !!!

Depends on:

* https://github.com/spacetelescope/jdaviz/pull/1481

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [x] Is a milestone set?
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/core/helpers.py
Comment (duytnguyendtn) [reply jdaviz/core/helpers.py:333]: kronk-kronk-oh-yeah

**Review by kecnry — COMMENTED** [review]
This is going to be great to have!  To be consistent with the syntax for sidecar, how about "popout:window" and "popout:tab" where "popout" defaults to one of those?  Or do you think that's too... specific to the underlying implementation as opposed to the actual location?

Whatever you decide, the API docs also definitely need updating.

**Comment by duytnguyendtn** [comment]
> do you think that's too... specific to the underlying implementation

No, I think that's excellent! Honestly, I'm starting to think `sidecar:X` might be too specific to the implementation ("sidecar" doesn't carry any specific meaning in my head at least

**Comment by duytnguyendtn** [comment]
The test failure is coming from a stale test that is already removed in https://github.com/spacetelescope/jdaviz/pull/1481. The tests should pass once we merge #1481 and I rebase this PR on main

**Review by kecnry — APPROVED** [review]
Works and I like the simplified syntax!  Needs a rebase before merge, but I'll approve anyways since its just the changelog that is conflicting with the latest release.

Future idea: it might be nice to eventually be able to pass size options and default to non full-screen (at least that's what it defaults to on my machine/browser), but that's out of scope here.  In theory, we can do that with pixel units by defaulting to `app.popout_button = PopoutButton(self, window_features='popup,width=800,height=1000')` or by changing `app.popout_button.window_features` before calling `app.popout_button.open_window/tab()`, but it would take a bit more logic to have reasonable screen-resolution-based defaults.

**Review by pllim — COMMENTED** [review]
Minor doc nitpicks. Otherwise, LGTM. Thanks!

File: CHANGES.rst
Comment (duytnguyendtn) [reply CHANGES.rst:7]: Would that overbloat the changelog? I figured if the user wanted to learn more, they could see the docs?

Comment (duytnguyendtn) [reply CHANGES.rst]: Nevermind! Doesn't look too crowded. Implemented in https://github.com/spacetelescope/jdaviz/pull/1503/commits/9abb185c04af3e839c8b47ab2a21ad6169c94ce9

File: docs/display.rst
Comment (pllim) [inline docs/display.rst:24]: Need the blank line.
```suggestion
Jdaviz can also be displayed in a detached window, separate from your working Jupyter interface.

.. note:: Popups must be allowed in your browser to display properly.
```

File: CHANGES.rst
Comment (pllim) [inline CHANGES.rst:8]:
```suggestion
  or browser tab (``popout:tab``). [#1503]
```

File: jdaviz/core/helpers.py
Comment (pllim) [inline jdaviz/core/helpers.py:308]:
```suggestion
            window will open. Browser popup permissions required.
```

**Review by pllim — COMMENTED** [review]
Nitpicks, since you have to fix RST anyway. 😅

**Comment by duytnguyendtn** [comment]
Whoops, didn't realize that noteblock wasn't rendering properly! Fixed it at least on my end

**Review by pllim — APPROVED** [review]
LGTM now. Thanks for your patience! I'll merge when CI passes.

**Comment by duytnguyendtn** [comment]
Thanks for the eyes!

**Pull request merged**
